compiler-plugin/src/main/kotlin/arrow/meta/quotes/Parameter.kt

```diff
@@ -0,0 +1,31 @@
+package arrow.meta.quotes
+
+import arrow.meta.Meta
+import arrow.meta.phases.ExtensionPhase
+import org.jetbrains.kotlin.name.Name
+import org.jetbrains.kotlin.psi.*
+import org.jetbrains.kotlin.psi.psiUtil.modalityModifierType
+import org.jetbrains.kotlin.psi.psiUtil.visibilityModifierType
+
+/**
+ * A [KtParameter] [Quote] with a custom template destructuring [ParameterScope]
+ */
+fun Meta.parameter(
+  match: KtParameter.() -> Boolean,
+  map: ParameterScope.(KtParameter) -> Transform<KtParameter>
+) : ExtensionPhase =
+  quote(match, map) { ParameterScope(it) }
+
+/**
+ * A template destructuring [Scope] for a [KtParameter]
+ */
+class ParameterScope(
+  override val value: KtParameter,
+  val modality: Name? = value.modalityModifierType()?.value?.let(Name::identifier),
+  val visibility: Name? = value.visibilityModifierType()?.value?.let(Name::identifier),
+  val name: Name? = value.nameAsName,
+  val `@annotationEntries`: ScopedList<KtAnnotationEntry> = ScopedList(value.annotationEntries),
```

Comment:
@raulraja why is `@annotationEntries` in string format? I've seen this else where in the quote system but I was wondering if there is more context about it

Reply:
When structuring a template you want the member to look as close to the parts:
```
`@annotations`
`(valueArguments)`
`(typeArguments)`
```
For example:
```kotlin
"""|$`@annotation` $kind $name $`(typeParameters)` $`(valueParameters)` : $supertypes {
   |  $body
   |}
   |""".`class`.synthetic
```

Reply:
symbols like `@` `(` and `)` are not allowed in Kotlin unless quoted. Alternatively, we can offer the same version of those without the quotes and the special symbols and let people use whichever looks more idiomatic for their use case

Reply:
@bloderxd I think that we actually need to remove `@annotationEntries` as well. This is not an element that is part of `KtParameter`.  The same goes for `modality` and `visibility`

Reply:
@ahinchman1 `KtParameter` inherits `KtAnnotated` which is in most elements that receive annotations including parameters